### PR TITLE
Fix uncrustify#3910: sp_after_sparen doesn't reflect settings

### PR DESCRIPTION
**Pull request by SebastianBilek**
This is a response to the issue that I opened up here: [https://github.com/uncrustify/uncrustify/issues/3910](https://github.com/uncrustify/uncrustify/issues/3910)
Basically what I did was change the order around because some default operations were preventing the actual functions from running and I ended up fixing the way IARF_ADD events are handled in that now it doesn't have trouble with CT_VBRACE_OPEN tokens.
Also I had some test problems that I didn't consider before so now those should be fixed up.

### CODE REVIEW

**Comment by SebastianBilek** [comment]
More problems...  :(

**Comment by SebastianBilek** [comment]
fixed those problems :)

**Comment by SebastianBilek** [comment]
Hi there, I had problems passing the test_cli_options.py test and sort of just assumed that --apply would help.
Could you please explain to me how I can compile with RELEASE?
Thanks, Seb.

File: tests/cli/output/universalindent.cfg [hunk 1]
```diff
@@ -777,12 +777,12 @@ ValueDefault=ignore
...
 [Sp After Sparen]
 Category=1
-Description="<html>Add or remove space after ')' of control statements.</html>"
+Description="<html>Add or remove space after ')' of control statements.<br/><br/>Default: force</html>"
 Enabled=false
 EditorType=multiple
 Choices=sp_after_sparen=ignore|sp_after_sparen=add|sp_after_sparen=remove|sp_after_sparen=force|sp_after_sparen=not_defined
 ChoicesReadable="Ignore Sp After Sparen|Add Sp After Sparen|Remove Sp After Sparen|Force Sp After Sparen"
-ValueDefault=ignore
+ValueDefault=force
```
Comment (micheleCTDEAdmin) [inline tests/cli/output/universalindent.cfg:785]: Default should be ignore like all the other options.

File: tests/cli/output/tracking_space.html [hunk 2]
```diff
@@ -7,17 +7,17 @@
...
-}
-template<typename ...A, int... B>
-struct foo1 : foo1<A..., (sizeof...(A) + B)...>
-{
-   foo1() {
-      int x = sizeof...(A);
-      bool b = x > 1;
-   }
-};
+{<a title="REMOVE: 1"><font color="red">M</font></a>
```
Comment (micheleCTDEAdmin) [inline tests/cli/output/tracking_space.html:10]: This file should remain unchanged

**Review by micheleCTDEAdmin — CHANGES_REQUESTED** [review]
Please check the provided feedback and adjust accordingly.

**Comment by micheleCTDEAdmin** [comment]
> Hi there, I had problems passing the test_cli_options.py test and sort of just assumed that --apply would help. Could you please explain to me how I can compile with RELEASE? Thanks, Seb.

Hi @SebastianBilek,
the readme file explains how to build with release or debug :-)

**Comment by SebastianBilek** [comment]
Hi @micheleCTDEAdmin, Sorry for that amateur question - I don't know how I missed that.
My reasoning for defaulting the sp_after_sparen setting to force is that before I applied my changes the sp_after_sparen setting it acted like it always was force - no matter the setting, so I wasn't sure whether you would be alright with me correcting all your tests to ignoring sp_after_sparen - which none of them do.

File: tests/cli/output/tracking_space.html (hunk 2 above)
Comment (SebastianBilek) [reply tests/cli/output/tracking_space.html]: Yep this is done.

**Comment by gmaurel** [comment]
the default value should be "ignore"

**Comment by SebastianBilek** [comment]
So, has it been decided?
If so, may I do it automatically or should I manually modify all the tests?

**Comment by micheleCTDEAdmin** [comment]
yes, both @guy-maurel and I think the default should be `ignore`. Please update the PR accordingly.

File: src/space.cpp [hunk 3]
```diff
@@ -17,12 +17,11 @@
  * @license GPL v2+
  */
 
-#include "space.h"
-
 #include "add_space_table.h"
 #include "log_rules.h"
 #include "options_for_QT.h"
 #include "punctuators.h"
+#include "space.h"
```
Comment (gmaurel) [inline src/space.cpp:24]: Leave the include at the old place, before the block of many includes.

**Review by gmaurel — CHANGES_REQUESTED** [review]
keep the old include place

File: src/space.cpp (hunk 3 above)
Comment (SebastianBilek) [reply src/space.cpp:24]: Yep got it, sorry for that :sweat_smile:

**Review by micheleCTDEAdmin — APPROVED** [review]
Looks good to me. Still need @guy-maurel approval before merging.

**Review by gmaurel — APPROVED** [review]

**Comment by micheleCTDEAdmin** [comment]
Thanks @SebastianBilek

**Pull request merged**